src/ethereum/ethEngine.js

```diff
@@ -0,0 +1,947 @@
+        const ethereumFee: EthereumFee = fees['default']
+        if (!ethereumFee.gasPrice) {
+          return
+        }
+        const gasPrice: EthereumFeesGasPrice = ethereumFee.gasPrice
+
+        const safeLow = Math.floor(jsonObj.safeLow / 10)
+        let average = Math.floor(jsonObj.average / 10)
+        let fastest = Math.floor(jsonObj.fastest / 10)
+
+        // Sanity checks
+        if (safeLow < 1 || safeLow > 300) {
+          console.log('Invalid safeLow value from EthGasStation')
+          return
+        }
+        if (average < 1 || average > 300) {
+          console.log('Invalid average value from EthGasStation')
+          return
+        }
+        if (fastest < 1 || fastest > 300) {
+          console.log('Invalid fastest value from EthGasStation')
+          return
+        }
+
+        const lowFee = (safeLow * 1000000000).toString()
+
+        if (average <= safeLow) average = safeLow + 1
+        const standardFeeLow = (average * 1000000000).toString()
+
+        if (fastest <= average) fastest = average + 1
+        const highFee = (fastest * 1000000000).toString()
+
+        // We use a value that is somewhere in between average and fastest for the standardFeeHigh
+        const standardFeeHigh = (
+          Math.floor((average + fastest) * 0.75) * 1000000000
+        ).toString()
+
+        if (
+          gasPrice.lowFee !== lowFee ||
+          gasPrice.standardFeeLow !== standardFeeLow ||
+          gasPrice.highFee !== highFee ||
+          gasPrice.standardFeeHigh !== standardFeeHigh
+        ) {
+          gasPrice.lowFee = lowFee
+          gasPrice.standardFeeLow = standardFeeLow
+          gasPrice.highFee = highFee
+          gasPrice.standardFeeHigh = standardFeeHigh
+          this.walletLocalDataDirty = true
+        }
+      } else {
+        this.log('Error: Fetched invalid networkFees from EthGasStation')
+      }
+    } catch (err) {
+      this.log('Error fetching networkFees from EthGasStation')
+      this.log(err)
+    }
+  }
+
+  async clearBlockchainCache () {
+    await super.clearBlockchainCache()
+    this.otherData.nextNonce = '0'
+  }
+
+  // ****************************************************************************
+  // Public methods
+  // ****************************************************************************
+
+  async startEngine () {
+    this.engineOn = true
+    this.addToLoop('checkBlockchainInnerLoop', BLOCKCHAIN_POLL_MILLISECONDS)
+    this.addToLoop('checkAccountInnerLoop', ACCOUNT_POLL_MILLISECONDS)
+    this.addToLoop('checkUpdateNetworkFees', NETWORKFEES_POLL_MILLISECONDS)
+    this.addToLoop('checkTransactionsInnerLoop', TRANSACTION_POLL_MILLISECONDS)
+    this.addToLoop('checkUnconfirmedTransactionsInnerLoop', UNCONFIRMED_TRANSACTION_POLL_MILLISECONDS)
+    super.startEngine()
+  }
+
+  async killEngine () {
+    // Set status flag to false
+    this.engineOn = false
+    // Clear Inner loops timers
+    for (const timer in this.timers) {
```

Comment:
The logic for killing timers seems pretty similar between plugins. Might make sense to put in the base class.

Reply:
Good catch. moved killEngine to the base class